### ISSUE TITLE
Add support for boot_mode option for Amazon EBS builder

### ISSUE THREAD

**Issue by hegyre** [hashicorp/packer-plugin-amazon#465]
#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

Those PRs #131 and #340 add support for boot_mode, but only for the "Chroot" and "EBS Surrogate" builders.
I would need to be able to select the boot_mode also for "EBS" builder.

#### Use Case(s)

[CIS hardening](https://www.tenable.com/audits/items/CIS_Debian_Linux_10_v1.0.0_L2_Server.audit:2670236182e11f7e8a5d7fd74048994b) needs to disable vfat filesystem. However, as mentionned:
> Disabling the vfat module can prevent boot on UEFI systems.

I'd like to test it out by creating an AMI while booting with Legacy-bios instead of UEFI.

#### Potential configuration

Same as documented on the [EBS Surrogate builder documentation](https://developer.hashicorp.com/packer/integrations/hashicorp/amazon/latest/components/builder/ebssurrogate) for `boot_mode` and optionnaly for `uefi_data`.
Valid options should be `legacy-bios` and `uefi` + the non-mentionned `uefi-preferred`.

#### Potential References
#131 and #340

**Comment by hegyre** [comment]
Hello,
It's not for creating the image itself but for running the temporary EC2 instance.

**Comment by lorengordon** [comment]
Ahh, the BootMode is inherinted from the selected source image. You cannot set it in the [RunInstances API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html).

You can force BIOS mode by using a UEFI Preferred image, and selecting an older instance type, say an m4, that does not support Nitro (required for UEFI).

**Comment by lorengordon** [comment]
Yeah, the API support for BootMode is very limited. None of CreateImage, CopyImage, or RunInstances support setting the Boot Mode. It can only be set by RegisterImage. If you have paid AWS Support, it would probably be worth opening a feature request to enhance the API support for BootMode.
